### PR TITLE
Merge gz-transport13 ➡️  main

### PR DESCRIPTION
**Pull request by scpeters**
# ➡️  Forward port

Port `gz-transport13 ` ➡️  `main`

Branch comparison: https://github.com/gazebosim/gz-transport/compare/main...gz-transport13

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: src/cmd/gz_TEST.cc [hunk 1]
```diff
@@ -304,23 +304,26 @@ TEST(gzTest, TopicPublish)
     "-t", "/bar",
     "-m", "gz.msgs.__bad_msg_type",
     "-p", R"(data: "good_value")"});
-  EXPECT_EQ(output.cerr.compare(0, error.size(), error), 0);
+  EXPECT_EQ(output.cerr.compare(0, error.size(), error), 0)
```
Comment (mjcarroll) [inline src/cmd/gz_TEST.cc:307]: Just a note that I have caught a few tests where we previously expected something on cout was now on cerr or vice-versa, but we only parsed the combined output before.

Comment (scpeters) [reply src/cmd/gz_TEST.cc:307]: I'm seeing a lot of these failures in `UNIT_gz_TEST` are caused by the following unexpected console output on Ubuntu 22.04 CI: `Library error: libgz-tools2-backward.so.2.0.0 not found. Improved backtrace generation will be disabled`. I believe it is a consequence of gazebosim/gz-tools#133 and think I know how to fix it

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

File: src/cmd/gz_TEST.cc (hunk 1 above)
Comment (scpeters) [reply src/cmd/gz_TEST.cc:307]: I updated the debian metadata in https://github.com/gazebo-release/gz-tools2-release/pull/15 and new nightlies have been rebuilt and the tests are now passing

**Comment by scpeters** [comment]
looks like CI is happy now, aside from coverage

File: README.md [hunk 2]
```diff
@@ -9,10 +9,10 @@
 
 Build | Status
 -- | --
-Test coverage | [![codecov](https://codecov.io/gh/gazebosim/gz-transport/branch/main/graph/badge.svg)](https://codecov.io/gh/gazebosim/gz-transport/branch/main)
-Ubuntu Focal | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_transport-ci-main-focal-amd64)](https://build.osrfoundation.org/job/ignition_transport-ci-main-focal-amd64)
-Homebrew      | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_transport-ci-main-homebrew-amd64)](https://build.osrfoundation.org/job/ignition_transport-ci-main-homebrew-amd64)
-Windows       | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ign_transport-ci-win)](https://build.osrfoundation.org/job/ign_transport-ci-win/)
+Test coverage | [![codecov](https://codecov.io/gh/gazebosim/gz-transport/tree/gz-transport13/graph/badge.svg)](https://codecov.io/gh/gazebosim/gz-transport/tree/gz-transport13)
```
Comment (azeey) [inline README.md:12]: The branch names should still be `main`.

File: src/cmd/gz_TEST.cc [hunk 3]
```diff
@@ -506,110 +372,100 @@ TEST(gzTest, TopicEcho)
...
-
-  EXPECT_GT(flags.size(), 0u);
-
-  // Match each flag in script output with help message
-  for (const auto &flag : flags)
-  {
-    EXPECT_NE(std::string::npos, helpOutput.find(flag)) << helpOutput;
-  }
-}
+// TEST(gzTest, ServiceHelpVsCompletionFlags)
```
Comment (azeey) [inline src/cmd/gz_TEST.cc:406]: I was surprised to see this, but https://github.com/gazebosim/gz-transport/pull/478/files#r1480280599

Comment (scpeters) [reply src/cmd/gz_TEST.cc:406]: ok, I had been waiting on this PR to get https://github.com/gazebosim/gz-transport/pull/472 merged to `main` to fix some CI issues, but I'll just cherry-pick that while we resolve these test issues

File: README.md (hunk 2 above)
Comment (iche033) [reply README.md]: fixed in [9aaa434](https://github.com/gazebosim/gz-transport/pull/479/commits/9aaa43440fb3bfbe4adcdbac1dce503a4e95c97c)

**Review by caguero — APPROVED** [review]

**Comment by iche033** [comment]
@caguero is the `playback.ReplayStep` test failure on windows expected / flaky?

**Comment by caguero** [comment]
> @caguero is the `playback.ReplayStep` test failure on windows expected / flaky?

I think it's flaky, I've seen it fail before.

**Review by iche033 — APPROVED** [review]

**Pull request merged**